src/sentry/release_health/metrics_sessions_v2.py

```diff
@@ -353,31 +434,103 @@ def run_sessions_query(
         elif ["session.status"] == query.raw_groupby:
             for status in SessionStatus:
                 # Create entry in default dict:
-                output_groups[GroupKey(session_status=status.value)]
-
-    # Convert group keys back to dictionaries:
-    results["groups"] = [
-        {"by": group_key.to_output_dict(), **group} for group_key, group in output_groups.items()  # type: ignore
-    ]
+                output_groups[GroupKey(session_status=status)]
+
+    return {
+        "groups": [
+            # Convert group keys back to dictionaries:
+            {"by": group_key.to_output_dict(), **group}  # type: ignore
+            for group_key, group in output_groups.items()
+        ],
+        "start": isoformat_z(metrics_results["start"]),
+        "end": isoformat_z(metrics_results["end"]),
+        "intervals": [isoformat_z(ts) for ts in metrics_results["intervals"]],
+        "query": query.query,
+    }
 
-    # Finally, serialize timestamps:
-    results["start"] = isoformat_z(results["start"])
-    results["end"] = isoformat_z(results["end"])
-    results["intervals"] = [isoformat_z(ts) for ts in results["intervals"]]
-    results["query"] = results.get("query", "")
 
-    return cast(SessionsQueryResult, results)
+def _empty_result(query: QueryDefinition) -> SessionsQueryResult:
+    intervals = get_timestamps(query)
+    return {
+        "groups": [],
+        "start": intervals[0],
+        "end": intervals[-1],
+        "intervals": intervals,
+        "query": query.query,
+    }
 
 
-def _get_filter_conditions(conditions: Sequence[Condition]) -> Any:
+def _get_filter_conditions(conditions: Any) -> ConditionGroup:
     """Translate given conditions to snql"""
     dummy_entity = EntityKey.MetricsSets.value
     return json_to_snql(
         {"selected_columns": ["value"], "conditions": conditions}, entity=dummy_entity
     ).where
 
 
-def _parse_orderby(query: QueryDefinition) -> Optional[OrderBy]:
+def _transform_conditions(
+    conditions: ConditionGroup,
+) -> Tuple[ConditionGroup, StatusFilter]:
+    """Split conditions into metrics conditions and a filter on session.status"""
+    if not conditions:
+        return conditions, None
+    where, status_filters = zip(*map(_transform_single_condition, conditions))
+    where = [condition for condition in where if condition is not None]
+    status_filters = [f for f in status_filters if f is not None]
+    if status_filters:
+        status_filters = frozenset.intersection(*status_filters)
```

Comment:
If I understand this correctly, the goal of this is to remove duplicates in the `status_filters` list. Doesn't `frozenset` do that by default?